servers/rendering/rendering_device.cpp

```diff
@@ -6131,6 +6717,8 @@ RenderingDevice::RenderingDevice() {
 	if (singleton == nullptr) {
 		singleton = this;
 	}
+
+	render_thread_id = Thread::get_caller_id();
```

Comment:
I think this should be captured in `initialize()` instead. Otherwise local rendering devices have to submit from the main rendering thread. Its fine to limit local rendering devices to a single thread, but they should live on the thread that created them, not on the main rendering thread if at all possible

Reply:
Discussed on RC. I misunderstood when the constructor is called. This already does exactly what I want.